### PR TITLE
set conccurency according to cpu resource limit/request if it is not set

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
fixes: #11268

### CODE REVIEW

File: pilot/pkg/kube/inject/concurrency.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+func applyConcurrency(containers []corev1.Container) {
+	for i, c := range containers {
+		if c.Name == ProxyContainerName {
+			concurrency := extractConcurrency(&c)
+			// do not change it when it is already set
+			if concurrency > 0 {
+				return
+			}
+
+			requestCPUsFloat := float64(c.Resources.Requests.Cpu().MilliValue()) / 1000
```
Comment (mandarjog) [inline pilot/pkg/kube/inject/concurrency.go]: If limit is specified it to should limit.
If request is 2 cpu and limit 4 we don’t want concurrency to be 2

Comment (hzxuzhonghu) [reply pilot/pkg/kube/inject/concurrency.go]: Correct

Comment (mandarjog) [reply pilot/pkg/kube/inject/concurrency.go]: I think I am missing something. This code is just looking at request .  *not* limit

Comment (mandarjog) [reply pilot/pkg/kube/inject/concurrency.go]: Lg re limits.

Nit: factor out 
‘UpdateConcurrencyIfSet(cpumillis) bool’
There is some specific code to update args there.

Comment (hzxuzhonghu) [reply pilot/pkg/kube/inject/concurrency.go]: done

**Comment by wenchenglu** [comment]
@hzxuzhonghu 
is #11268 for 1.1? if true please sync up with @duderino to track it as 1.1

**Comment by hzxuzhonghu** [comment]
We are urgent for this feature

**Comment by rshriram** [comment]
@wenchenglu yes.. this is not a feature but more of a fix as till date we have been making the proxies consume an unbounded amount of CPU. Using the pod's CPU budget as a proxy for the number of threads to launch in a sidecar, is a nice approximation. Without this fix, the amount of memory usage at every proxy increases by the number of threads in that proxy. And each thread consumes somewhere around 10Mb+ due to all the clusters, and their stats.

**Pull request merged**